From 26a31c744ef0c5fd9bb0c0dc948723b02f159660 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Thu, 8 Feb 2018 01:19:52 +0100
Subject: [PATCH 1/3] msm8974-common: libril: Fix RIL_Call struct for 64-bit
 platforms

* While msm8974 is not 64 bit, certain users of this libril will be
* A previous revision of this libril had 16 bytes of padding in this
  struct for 64-bit platforms (due to word size and struct alignment),
  but a change by me to use a long instead of a pointer and an int
  broke this padding for 64-bit platforms because a long is still
  just 8 bytes on 64-bit platforms, where 16 bytes is necessary
* Use two pointers as padding instead to make 64-bit great again

Change-Id: If599bdb625f7a45e083f5b30df512d12810be79b
---
 ril/include/telephony/ril.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ril/include/telephony/ril.h b/ril/include/telephony/ril.h
index 9c935aa..734a12b 100644
--- a/ril/include/telephony/ril.h
+++ b/ril/include/telephony/ril.h
@@ -436,7 +436,8 @@ typedef struct {
     int             numberPresentation; /* 0=Allowed, 1=Restricted, 2=Not Specified/Unknown 3=Payphone */
     char *          name;       /* Remote party name */
     int             namePresentation; /* 0=Allowed, 1=Restricted, 2=Not Specified/Unknown 3=Payphone */
-    long            call_details; /* Samsung call detail additions. Just padding, do not use */
+    void *          call_details1; /* Padding for Samsung's call detail fields */
+    void *          call_details2; /* Padding for Samsung's call detail fields */
     RIL_UUS_Info *  uusInfo;    /* NULL or Pointer to User-User Signaling Information */
 } RIL_Call;
 
-- 
2.7.4

